src/constraints/StringConstraints.ts

```diff
@@ -44,3 +59,68 @@ export function stringLengthNe(length: number): IConstraint<string> {
 	const expected = `expected.length !== ${length}`;
 	return stringLengthComparator(ne, 's.string.lengthNe', expected, length);
 }
+
+export function stringRegex(regex: RegExp, type: 'url' | 'uuid' | 'regex' | 'email'): IConstraint<string> {
+	const expected = `expected to match ${type === 'regex' ? regex.source : `an ${type}`}`;
```

Comment:
Bleh, the only one that needs the `n` in `an` is email, everything else looks out of place... @kyranet @favna should we just bite the bullet on `a email` or...